static/app/utils/discover/eventView.tsx

```diff
@@ -609,6 +599,13 @@ class EventView {
       delete newQuery.query;
     }
 
+    if (this.dataset) {
+      newQuery.queryDataset = getSavedQueryDatasetFromLocationOrDataset(
+        undefined,
+        this.dataset
+      );
+    }
```

Comment:
I'm a little confused about why we need to pass along the `queryDataset` field to a new query if these changes were meant to remove this, could `EventView` just handle a single `dataset` field? Is this being used?

Reply:
I went back and forth on this a bit, `toNewQuery()` is primarily used to create the payload for creating/updating a DiscoverSavedQuery object. So I figured I'd rather implement it here than at every individual discover saved query api call. When we implement this change for dashboards, we could add another helper method to EventView like `toNewWidgetQuery()` and do something similar.

Reply:
I see, yeah, I think if this is supposed to produce the most valid instance of a `DiscoverSavedQuery` object then I'd rather us keep the logic together than having to remember to add the `queryDataset` attribute everywhere it's called. Thanks for clarifying why this is here!